R/teal_data-extract.R

```diff
@@ -0,0 +1,38 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
+#' @param x (`teal_data`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' data <- teal_data()
+#' data <- eval_code(data, "a <- 1;b<-2")
+#' data["a"]
+#' data[c("a", "b")]
+#'
+#' join_keys(data) <- join_keys(join_key("a", "b", "x"))
+#' join_keys(data["a"]) # should show empty keys
+#' join_keys(data["b"])
+#' join_keys(data)["a"] # should show empty keys
+#' join_keys(data)["b"]
+#'
+#' @rdname teal_data
+#'
+#' @export
+`[.teal_data` <- function(x, names) {
+  checkmate::assert_class(names, "character")
+  names <- intersect(names, ls(get_env(x)))
```

Comment:
Should extract of non-existing index throw a warning?

Reply:
Unsure, maybe it doesn't hurt to include such a warning! Let's include this

Reply:
should we throw a warning at all?

I'm looking at this like a subsetting from env:
```
r$> e <- list2env(list(a = 1, b = 2))

r$> e[['a']]
[1] 1

r$> e[['aa']]
NULL
```
I see no warning in this.

Reply:
@pawelru Marcin is implementing `[` instead of `[[`. I proposed a warning because `[` returns strange results for inexisting indexes which can't be translated one-to-one to `teal_data` object:

```r
e <- list(a = 1, b = 2) # list because environment has no `[` method
e[NULL]
# named list()

e[character(0)]
# named list()
e["c"]

e[]
# $a
# [1] 1
# 
# $b
# [1] 2

e["inexisting"]
# $<NA>
# NULL
```

We can discuss whether to warn or not here, but this should be implemented in `[.qenv`